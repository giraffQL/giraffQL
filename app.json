{
    "name": "GiraffQL",
    "description": "Interactive GraphQL exploration tool built with React",
    "repository": "https://github.com/giraffQL/giraffQL.git"
}