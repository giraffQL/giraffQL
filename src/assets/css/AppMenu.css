.appBar {
  width:100%;
  height: 65px;
  display:flex;
  justify-content: space-between;
  align-items: center;
  color: white;
}

/*add and delete buttons*/
.addAndClearBtns {
  float: left;
  color: rgb(51,51,51);
  font-weight: bold;
  border: 1px solid rgb(45,45,45);
  border-radius: 50%;
  background-color: #FFD000;
  margin-right: 7px;
}

#createTableBtn {
  /*color: white;*/
  background-color: rgba(30,130,76,1);
}

#createTableBtn:hover {
  color: rgba(30,130,76,1);
  background-color: white;
  border-color: white;
}

#clearBtn {
  background-color: rgba(166,0,0,1);
}

#clearBtn:hover {
  color: rgba(166,0,0,1);
  background-color: white;
  border-color: white;
}

.save {
  color: rgb(51,51,51);
  background-color: #fdd217; /*#FFD000;*/
  font-size: 18px;
  font-weight: bold;
  padding: 10px;
  margin-right:10px;
}

.save:hover {
  background-color: white;
  color: rgb(51,51,51);
}

.test {
  color: rgb(51,51,51);
  background-color: #fdd217; /*#FFD000;*/
  font-size: 18px;
  font-weight: bold;
  padding: 10px;
  margin-right:10px;
}

.test:hover {
  background-color: white;
  color: rgb(51,51,51);
}

#tooltip {
  font-size: 20px;
}