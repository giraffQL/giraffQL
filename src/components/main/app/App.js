import React, { Component } from 'react';
import SplitPane from "react-split-pane"
import Fullscreen from 'react-full-screen';
// MATERIAL UI
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import FlatButton from 'material-ui/FlatButton';
// REACT-BOOTSTRAP
import { FormControl, Button, ButtonGroup, Nav, Popover } from 'react-bootstrap';
//FILE SERVER
import FileSaver from 'file-saver';
//COMPONENTS
import MenuComp from './AppMenu';
import Visualization from './table/Visualization';
import TextEditor from './code/TextEditor'
import ExpressCode from './code/ExpressCode'





class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      clickedRow: null,
      // our data have tables which is object of tables
      // {
      //   id: '1',
      //   name: 'One',
      //   tablePositionX: 0,
      //   tablePositionY: 7,
      //   attributes: [
      //     { field: 'hi', type: '', relatedToTableId: '2' },
      //     { field: 'blah', type: '', relatedToTableId: null }
      //   ]
      // }
      // each table has these properties...attributes are rows and each row has field, type and table which
      // is related to
      data: {
        tables: [
        ]
      },
      schemaCode: '',
      jsCode: ''
    };
  };

  // when we click to make new table call this function
  onAddTable = () => {
    //function which is making random string for ID
    function guid() {
      function s4() {
        return Math.floor((1 + Math.random()) * 0x10000)
          .toString(16)
          .substring(1);
      }
      return s4() + s4() + '-' + s4() + '-' + s4() + '-' +
        s4() + '-' + s4() + s4() + s4();
    }

    let newstate = this.state.data.tables.slice()
    const newData = {
      tables: newstate.concat({
        id: guid(),
        name: '',
        attributes: [
          { field: '', type: '' }
        ],
        defaultPosition: 210 * newstate.length
      })
    }
    this.setState({
      data: newData,
      schemaCode: this.getTextFromModel(newData),
      jsCode: this.getExpressCode(newData)
    })
  }

  // function which is calling when we're adding new row
  onAddRow = (tableIndex) => {
    this.setState(state => {
      const newData = {
        tables: state.data.tables.map((table, i) =>
          (i === tableIndex)
            ? Object.assign({}, table, {
              attributes: table.attributes.concat({ field: '', type: '' })
            })
            : table
        )
      }
      return {
        data: newData,
        schemaCode: this.getTextFromModel(newData),
        jsCode: this.getExpressCode(newData)
      }
    })
  }

  // every time when we change or add table name we're setting up state with thoose new values
  updateTableName = (tableIndex, value) => {
    this.setState(state => {
      const newData = {
        tables: state.data.tables.map((table, i) =>
          (i === tableIndex)
            ? Object.assign({}, table, {
              name: value,
              attributes: table.attributes.map((attr, ia) =>
                (attr.relatedToTableId === state.data.tables[tableIndex].id)
                  ? Object.assign({}, attr, { type: value })
                  : attr
              )
            })
            : Object.assign({}, table, {
              attributes: table.attributes.map((attr, ia) =>
                (attr.relatedToTableId === state.data.tables[tableIndex].id)
                  ? Object.assign({}, attr, { type: value })
                  : attr
              )
            })
        )
      }
      return {
        data: newData,
        schemaCode: this.getTextFromModel(newData),
        jsCode: this.getExpressCode(newData)
      }
    })
  }
  // same for row prop and row type
  updateRowProp = (tableIndex, rowIndex, value) => {
    this.setState(state => {
      const newData = {
        tables: state.data.tables.map((table, i) =>
          (i === tableIndex)
            ? Object.assign({}, table, {
              attributes: table.attributes.map((attr, ai) =>
                (ai === rowIndex)
                  ? Object.assign({}, attr, { field: value })
                  : attr
              )
            })
            : table
        )
      }
      return {
        data: newData,
        schemaCode: this.getTextFromModel(newData),
        jsCode: this.getExpressCode(newData)
      }
    })
  }

  updateRowType = (tableIndex, rowIndex, value) => {
    this.setState(state => {
      const newData = {
        tables: state.data.tables.map((table, i) =>
          (i === tableIndex)
            ? Object.assign({}, table, {
              attributes: table.attributes.map((attr, ai) =>
                (ai === rowIndex)
                  ? Object.assign({}, attr, { type: value })
                  : attr
              )
            })
            : table
        )
      }
      return {
        data: newData,
        schemaCode: this.getTextFromModel(newData),
        jsCode: this.getExpressCode(newData)
      }
    })
  }

  // function which is calling when we click for delete row
  deleteRow = (tableIndex, rowIndex) => {
    this.setState(state => {
      const newData = {
        tables: state.data.tables.map((table, i) =>
          (i === tableIndex)
            ? Object.assign({}, table, {
              attributes: table.attributes.filter((attr, ai) => ai !== rowIndex)
            })
            : table
        )
      }
      return {
        data: newData,
        schemaCode: this.getTextFromModel(newData),
        jsCode: this.getExpressCode(newData)
      }
    })
  }

  // function which is calling when we click for delete table
  deleteTable = (tableIndex) => {
    this.setState(state => {
      const newData = {
        tables: state.data.tables.filter((table, i) => i !== tableIndex)
      }
      return {
        data: newData,
        schemaCode: this.getTextFromModel(newData),
        jsCode: this.getExpressCode(newData)
      }
    })
  }

  // function which is calling when we click for clear board
  deleteAllTables = () => {
    this.setState(state => {
      const newData = {
        tables: []
      }
      return {
        data: newData,
        schemaCode: this.getTextFromModel(newData),
        jsCode: this.getExpressCode(newData)
      }
    })
  }

  // function which is calling when we drag tables
  // every time when we drag table we are keeping track of table position, and row position
  refreshTablePositions = (tableIndex, tablePosition, rowPositions) => {
    this.setState(state => {
      //table
      let table = state.data.tables[tableIndex]
      if (table) {
        table.tablePositionX = table.x = Math.floor(tablePosition.left)
        table.tablePositionY = table.y = Math.floor(tablePosition.top)
        table.w = Math.floor(tablePosition.width)
        table.h = Math.floor(tablePosition.height)
        //rows
        let attrs = state.data.tables[tableIndex].attributes
        for (let i = 0; i < attrs.length; i += 1) {
          attrs[i].x = Math.floor(rowPositions[i].left)
          attrs[i].y = Math.floor(rowPositions[i].top)
          attrs[i].w = Math.floor(rowPositions[i].width)
          attrs[i].h = Math.floor(rowPositions[i].height)
        }
      }
      return state;
    })
  }

  // clickedRow is null at the beginning but when we click on row we are setting state to
  // know which row is clicked, in which table...
  onRowMouseDown = (tableIndex, rowIndex) => {
    this.setState({
      clickedRow: {
        tableIndex,
        rowIndex
      }
    })
  }

  // storing to which table is row connected
  onTableMouseUp = (tableIndex) => {
    const { clickedRow } = this.state
    if (tableIndex === null || !clickedRow || clickedRow.tableIndex === tableIndex) {
      this.setState({
        clickedRow: null
      })
    } else {
      this.setState(state => {
        const table = state.data.tables[state.clickedRow.tableIndex]
        table.attributes[state.clickedRow.rowIndex].relatedToTableId = state.data.tables[tableIndex].id
        table.attributes[state.clickedRow.rowIndex].type = state.data.tables[tableIndex].name
        return {
          clickedRow: null,
          data: state.data,
          schemaCode: this.getTextFromModel(state.data),
          jsCode: this.getExpressCode(state.data)
        }
      })
    }
  }


  menuToggle = () => {
    this.setState({ open: !this.state.open });
  }

  menuClose = () => {
    this.setState({ open: false });
  }

  onRequestChange = (open) => {
    this.setState({ open });
  }

  fullscreenToggle = () => {
    this.setState({
      isFullscreenEnabled: this.state.isFullscreenEnabled ? false : true,
      open: false
    })
  }

  // function which is called when you click for save schema js code
  saveTextAsFile = () => {
    let text = this.state.jsCode
    let blob = new Blob([text], { type: "text/javascript" });
    FileSaver.saveAs(blob, 'schema.js')
  }


  onSchemaCodeChange = (schemaCode) => {
    this.setState(state => ({
      schemaCode
    }))
  }

  onJsCodeChange = (jsCode) => {
    this.setState(state => ({
      jsCode
    }))
  }

  submitSchemaCode = () => {
    function post(path, params, method) {
      method = method || "post";

      let form = document.createElement("form");
      form.setAttribute("method", method);
      form.setAttribute("action", path);

      for (let key in params) {
        if (params.hasOwnProperty(key)) {
          let hiddenField = document.createElement("input");
          hiddenField.setAttribute("type", "hidden");
          hiddenField.setAttribute("name", key);
          hiddenField.setAttribute("value", params[key]);

          form.appendChild(hiddenField);
        }
      }
      document.body.appendChild(form);
      form.submit();
    }
    post('/schemas', { schema: this.state.schemaCode })
  }

  // function to check is input string contains number
  startsWithNumber = (text) => {
    return /^[0-9]/.test(text);
  }

  // function which is creating text in schema text area
  getTextFromModel = (data) => {
    let code = '\n'
    code += 'type Query {\n'

    for (let i = 0; i < data.tables.length; i += 1) {
      const table = data.tables[i]

      if (table.name && !this.startsWithNumber(data.tables[i].name)) {
        code += `    ${table.name}: ${table.name}\n`
      }
    }
    code += `}\n\n`

    for (let i = 0; i < data.tables.length; i += 1) {
      const table = data.tables[i]

      if (table.name && !this.startsWithNumber(data.tables[i].name)) {
        code += `type ${table.name} {\n`
        for (let j = 0; j < table.attributes.length; j += 1) {
          const attr = table.attributes[j]
          if (attr.field !== '') {
            code += `    ${attr.field}: ${attr.type}\n`
          }
        }
        code += `}\n\n`
      }
    }

    return code + '\n'
  }

  // function which is creating schema js code in textarea
  getExpressCode = (data) => {
    let code = '\n'
    for (let i = 0; i < data.tables.length; i += 1) {
      const table = data.tables[i]

      if (table.name && !this.startsWithNumber(data.tables[i].name)) {
        code += `const ${table.name}Type = new GraphQLObjectType({\n`
          + `    name: ${table.name},\n`
          + `    fields: () => ({\n`
        for (let j = 0; j < table.attributes.length; j += 1) {
          const attr = table.attributes[j]
          if (attr.field !== '') {
            code += `        ${attr.field}: {\n`
              + `            type: ${!attr.relatedToTableId ? 'GraphQL' : ''}${attr.type}${attr.relatedToTableId ? 'Type' : ''}\n`
              + `        }`
          }
          if (j < table.attributes.length - 1 && attr.field !== '') {
            code += `,\n`
          }
        }
        code += `\n`
          + `    })\n`
          + `})\n\n`
      }
    }

    return code + '\n'
  }

  render() {
    const { data } = this.state;
    return [
      <MuiThemeProvider>
        <div className="App">
          <Fullscreen style={{ 'height': '100%' }}
            enabled={this.state.isFullscreenEnabled}
            onChange={isFullscreenEnabled => this.setState({ isFullscreenEnabled })}
          >
            <div className='full-screenable-node'>
              <MenuComp state={this.state} menuToggle={this.menuToggle} menuClose={this.menuClose} onRequestChange={this.onRequestChange} fullscreenToggle={this.fullscreenToggle} onAddTable={this.onAddTable} deleteAllTables={this.deleteAllTables} saveTextAsFile={this.saveTextAsFile} submitSchemaCode={this.submitSchemaCode} />

              <SplitPane split="vertical" defaultSize="70%">
                {data.tables.length === 0 ?
                  <div className="firstShow">
                    <p className="textClick"> Click on <Button id="createTableBtn" bsSize="large" onClick={() => this.onAddTable()}>
                      +
                 </Button> to add table </p>
                  </div> :
                  <Visualization data={this.state.data} clickedRow={this.state.clickedRow} onAddRow={this.onAddRow} onAddTable={this.onAddTable}
                    updateTableName={this.updateTableName} updateRowProp={this.updateRowProp} updateRowType={this.updateRowType} onDragTable={this.onDragTable} refreshTablePositions={this.refreshTablePositions} deleteTable={this.deleteTable} deleteRow={this.deleteRow} deleteAllTables={this.deleteAllTables} onTableMouseUp={this.onTableMouseUp} onRowMouseDown={this.onRowMouseDown} />
                }
                <div className="TextEditor">
                  <div className="editorTxt">
                      Test Schema
                  </div>
                  <TextEditor code={this.state.schemaCode} onChange={this.onSchemaCodeChange} />
                  <div className="editorTxt">
                      Save Code
                  </div>
                  <ExpressCode code={this.state.jsCode} onChange={this.onJsCodeChange} />
                </div>
              </SplitPane>
            </div>
          </Fullscreen>
        </div>
      </MuiThemeProvider>
    ];
  }
}

export default App;

