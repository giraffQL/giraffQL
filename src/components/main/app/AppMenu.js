import React, { Component } from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import RaisedButton from 'material-ui/RaisedButton';
import AppBar from 'material-ui/AppBar';
import Drawer from 'material-ui/Drawer';
import MenuItem from 'material-ui/MenuItem';
import { Link } from 'react-router-dom'
import FlatButton from 'material-ui/FlatButton';
// REACT-BOOTSTRAP
import { FormControl, Button, ButtonGroup, Nav, Tooltip, OverlayTrigger } from 'react-bootstrap';


class MenuComp extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    const { state, menuToggle, menuClose, onRequestChange, fullscreenToggle, onAddTable, deleteAllTables, saveTextAsFile, submitSchemaCode } = this.props;

    const muiStyles = {
      appBar: {
        backgroundColor: 'black',
        lineHeight: '20px',
        borderBottom: '2px solid rgb(247,247,240)'
      },
      drawer: {
        backgroundColor: 'black',
      },
      menuItem: {
        color: '#fdd217',
        fontSize: '16px',
        fontWeight: '400',
        textAlign: 'center'
      },
      aboutUs: {
        color: '#fdd217',
        fontSize: '16px',
        fontWeight: '400',
        textAlign: 'center',
        borderBottom: '1px dashed #fdd217'
      },
      menuIcon: {
        'color': '#fdd217' /*'#FFD300'*/
      }
    }

    const tooltipCT = (
      <Tooltip id="tooltip">Create Table</Tooltip>
    )
    const tooltipCA = (
      <Tooltip id="tooltip">Clear All</Tooltip>
    )
    const tooltipTest = (
      <Tooltip id="tooltip"> Test schema with GraphiQL</Tooltip>
    )
    const tooltipSave = (
      <Tooltip id="tooltip"> Save express schema.js file </Tooltip>
    )
    return (
      <div>
        <div style={muiStyles.appBar} className="appBar">
          <FlatButton style={muiStyles.menuIcon} className='navbutton'
            onClick={menuToggle}
            icon={<i className="material-icons">list</i>}
          />
            <div className="toolPanel">
              <OverlayTrigger placement="left" overlay={tooltipCT}>
                <Button id="createTableBtn" className="addAndClearBtns" bsSize="default" onClick={onAddTable}>
                  +
                   </Button>
              </OverlayTrigger>
              <OverlayTrigger placement="bottom" overlay={tooltipCA}>
                  <Button id="clearBtn" className="addAndClearBtns" bsSize="default" onClick={deleteAllTables}>
                    x
                  </Button>
                  {/*<div className="btnDescription">
                    Clear All
                  </div>*/}
              </OverlayTrigger>
              <OverlayTrigger placement="bottom" overlay={tooltipTest}>
                  <button className="test" onClick={submitSchemaCode}>
                    TEST
                  </button>
              </OverlayTrigger>
              <OverlayTrigger placement="bottom" overlay={tooltipSave}>
              <button className="save" onClick={saveTextAsFile}>
                SAVE
              </button>
              </OverlayTrigger>
            </div>
        </div>
        <Drawer
          className='drawer'
          containerStyle={muiStyles.drawer}
          docked={false}
          width={200}
          open={state.open}
          onRequestChange={onRequestChange}
        >
        <div className ='drawertop' style={{'fontWeight': 'bold'}}>GiraffQL</div>
          <Link to="/">
            <MenuItem style={muiStyles.menuItem} onClick={menuClose}>
              Home
                </MenuItem>
          </Link>
          <Link to="/app">
            <MenuItem style={muiStyles.menuItem} onClick={menuClose}>
              App
                </MenuItem>
          </Link>
          <Link to="/aboutus">
            <MenuItem style={muiStyles.aboutUs} onClick={menuClose}>
              About us
                </MenuItem>
          </Link>
          <MenuItem style={muiStyles.menuItem} onClick={fullscreenToggle}>
            Fullscreen
              </MenuItem>
        </Drawer>
      </div>
    )
  }
}

export default MenuComp;