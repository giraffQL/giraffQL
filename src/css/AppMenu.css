.appBar {
  width:100%;
  display:flex;
  justify-content: space-between;
  align-items: center;
  color: white;
}

/*add and delete buttons*/
.addAndClearBtns {
  float: left;
  color: rgb(51,51,51);
  font-weight: bold;
  border: 2px solid rgb(51,51,51);
  border-radius: 50%;
  background-color: #FFD000;
}

#createTableBtn {
  /*color: white;*/
  background-color: rgba(30,130,76,1);
}

#createTableBtn:hover {
  color: rgba(30,130,76,1);
  background-color: white;
  border-color: white;
}

#clearBtn {
  background-color: rgba(166,0,0,1);
}

#clearBtn:hover {
  color: rgba(166,0,0,1);
  background-color: white;
  border-color: white;
}

.save {
  color: rgb(51,51,51);
  background-color: #FFD000;
  font-size: 18px;
  font-weight: bold;
  padding: 10px
}

.save:hover {
  background-color: white;
  color: rgb(51,51,51);
}

.test {
  color: rgb(51,51,51);
  background-color: #FFD000;
  font-size: 18px;
  font-weight: bold;
  padding: 10px
}

.test:hover {
  background-color: white;
  color: rgb(51,51,51);
}

#tooltip {
  font-size: 20px;
}