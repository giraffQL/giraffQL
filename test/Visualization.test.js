import React from 'react';
import ReactDOM from 'react-dom';
import App from '../src/components/main/app/App';
import AppMenu from '../src/components/main/app/AppMenu';
import Table from '../src/components/main/app/table/Table';
import Visualization from '../src/components/main/app/table/Visualization';
// import handleMouseMove from '../src/components/table/Visualization';
import { onAddTable } from '../src/components/main/app/App';
// import { Scrollbars } from 'react-custom-scrollbars';

import { shallow, mount, render } from "enzyme";

const data = { tables: [{ name: 'testName', attributes: [{ field: 'testField', type: 'testType', x: 0, y: 0 }, { field: 'testField2', type: 'testType2', x: 0, y: 0 }] }] };
const dataProps = {
  data: {
    tables: [
      { name: 'testName', attributes: [{ field: 'testField', type: 'testType', x: 0, y: 0 }, { field: 'testField2', type: 'testType2', x: 0, y: 0 }] },
      { name: 'testName', attributes: [{ field: 'testField', type: 'testType', x: 0, y: 0 }, { field: 'testField2', type: 'testType2', x: 0, y: 0 }] },
      { name: 'testName', attributes: [{ field: 'testField', type: 'testType', x: 0, y: 0 }, { field: 'testField2', type: 'testType2', x: 0, y: 0 }] }
    ]
  }
}
const mockState = {state: { open: 'test' }};

const jestMock = jest.fn();
const vis = shallow(<Visualization data={dataProps.data} onAddTable={jestMock} deleteAllTables={jestMock} onAddRow={jestMock} handleMouseMove={jestMock} />);
const visInst = vis.instance();
const app = shallow(<App />);
const bar = shallow(<AppMenu state={mockState} onAddTable={jestMock} deleteAllTables={jestMock} onAddRow={jestMock} />);

describe('Visualization Rendering Properly', () => {

  it('includes Create Table button', () => {
    expect(bar.find('#createTableBtn').length).toEqual(1);
  });

  it('includes Clear Table button', () => {
    expect(bar.find('#clearBtn').length).toEqual(1);
  });

  it('includes a table div', () => {
    expect(vis.find('.tables').length).toEqual(1);
  })
})

describe('Visualization Table Testing', () => {

  it('Vis renders correctly and match Snapshot', () => {
    expect(vis).toMatchSnapshot();
  });

  it('Calls onAddTable when button is clicked', () => {
    bar.find('#createTableBtn').simulate('click');
    expect(jestMock).toBeCalled();
  });

  it('Calls Clear when button is clicked', () => {
    bar.find('#clearBtn').simulate('click');
    expect(jestMock).toBeCalled();
  });

})

describe('Proper CSS Classes are rendered in visualization', () => {

  it('Scrollbars have class name of "toolbar"', () => {
    expect(vis.find('.toolbar').length).toEqual(1);
  });

  it('SVG to have class name of "relations"', () => {
    expect(vis.find('.relations').length).toEqual(1);
  });

  it('Tables div have class name of "tables"', () => {
    expect(vis.find('.tables').length).toEqual(1);
  });
  
  it('Contains class name of "relations"', () => {
    expect(vis.find('.relations').length).toEqual(1);
  });

  it('Contains class name of "visualization"', () => {
    expect(vis.find('.visualization').length).toEqual(1);
  });
  
})


describe('Functionality', () => {

  it('handleMouseMove fires when clicked', () => {
    vis.simulate('mouseMove');
    expect(jestMock).toHaveBeenCalled();
  });

  // it('handleMouseDown fires when clicked', () => {
  //   vis.simulate('mouseDown');
  //   expect(jestMock).toHaveBeenCalled();
  // });

  // it('handleMouseUp fires when clicked', () => {
  //   vis.simulate('mouseup');
  //   expect(jestMock).toHaveBeenCalled();
  // });

})