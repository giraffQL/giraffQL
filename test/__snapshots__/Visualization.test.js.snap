// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Visualization Table Testing Vis renders correctly and match Snapshot 1`] = `
<div
  className="visualization"
  onMouseDown={[Function]}
  onMouseMove={[Function]}
  onMouseUp={[Function]}
>
  <Scrollbars
    autoHeight={false}
    autoHeightMax={200}
    autoHeightMin={0}
    autoHide={false}
    autoHideDuration={200}
    autoHideTimeout={1000}
    hideTracksWhenNotNeeded={false}
    renderThumbHorizontal={[Function]}
    renderThumbVertical={[Function]}
    renderTrackHorizontal={[Function]}
    renderTrackVertical={[Function]}
    renderView={[Function]}
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
    tagName="div"
    thumbMinSize={30}
    universal={false}
  >
    <div
      className="toolbar"
    />
    <div>
      <svg
        className="relations"
      >
        <defs>
          <marker
            id="triangle"
            markerHeight="5"
            markerWidth="5"
            orient="auto"
            refX="1"
            refY="5"
            viewBox="0 0 10 10"
          >
            <path
              d="M 0 0 L 10 5 L 0 10 z"
              fill="#fdd217"
            />
          </marker>
          <marker
            id="circle"
            markerHeight="10"
            markerWidth="10"
            orient="auto"
            refX="5"
            refY="5"
          >
            <circle
              cx="5"
              cy="5"
              fill="#fdd217"
              r="2"
            />
          </marker>
        </defs>
        0
        0
        0
        0
        0
        0
      </svg>
      <div
        className="tables"
      >
        <Table
          data={
            Object {
              "tables": Array [
                Object {
                  "attributes": Array [
                    Object {
                      "field": "testField",
                      "type": "testType",
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "field": "testField2",
                      "type": "testType2",
                      "x": 0,
                      "y": 0,
                    },
                  ],
                  "name": "testName",
                },
                Object {
                  "attributes": Array [
                    Object {
                      "field": "testField",
                      "type": "testType",
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "field": "testField2",
                      "type": "testType2",
                      "x": 0,
                      "y": 0,
                    },
                  ],
                  "name": "testName",
                },
                Object {
                  "attributes": Array [
                    Object {
                      "field": "testField",
                      "type": "testType",
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "field": "testField2",
                      "type": "testType2",
                      "x": 0,
                      "y": 0,
                    },
                  ],
                  "name": "testName",
                },
              ],
            }
          }
          draggable={true}
          onAddRow={[Function]}
          onDragTable={[Function]}
          refreshTableRefs={[Function]}
          style={
            Object {
              "backgroundColor": "Safari",
            }
          }
          table={
            Object {
              "attributes": Array [
                Object {
                  "field": "testField",
                  "type": "testType",
                  "x": 0,
                  "y": 0,
                },
                Object {
                  "field": "testField2",
                  "type": "testType2",
                  "x": 0,
                  "y": 0,
                },
              ],
              "name": "testName",
            }
          }
          tableIndex={0}
          tables={
            Array [
              Object {
                "attributes": Array [
                  Object {
                    "field": "testField",
                    "type": "testType",
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "field": "testField2",
                    "type": "testType2",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "name": "testName",
              },
              Object {
                "attributes": Array [
                  Object {
                    "field": "testField",
                    "type": "testType",
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "field": "testField2",
                    "type": "testType2",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "name": "testName",
              },
              Object {
                "attributes": Array [
                  Object {
                    "field": "testField",
                    "type": "testType",
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "field": "testField2",
                    "type": "testType2",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "name": "testName",
              },
            ]
          }
        />
        <Table
          data={
            Object {
              "tables": Array [
                Object {
                  "attributes": Array [
                    Object {
                      "field": "testField",
                      "type": "testType",
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "field": "testField2",
                      "type": "testType2",
                      "x": 0,
                      "y": 0,
                    },
                  ],
                  "name": "testName",
                },
                Object {
                  "attributes": Array [
                    Object {
                      "field": "testField",
                      "type": "testType",
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "field": "testField2",
                      "type": "testType2",
                      "x": 0,
                      "y": 0,
                    },
                  ],
                  "name": "testName",
                },
                Object {
                  "attributes": Array [
                    Object {
                      "field": "testField",
                      "type": "testType",
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "field": "testField2",
                      "type": "testType2",
                      "x": 0,
                      "y": 0,
                    },
                  ],
                  "name": "testName",
                },
              ],
            }
          }
          draggable={true}
          onAddRow={[Function]}
          onDragTable={[Function]}
          refreshTableRefs={[Function]}
          style={
            Object {
              "backgroundColor": "Sienna",
            }
          }
          table={
            Object {
              "attributes": Array [
                Object {
                  "field": "testField",
                  "type": "testType",
                  "x": 0,
                  "y": 0,
                },
                Object {
                  "field": "testField2",
                  "type": "testType2",
                  "x": 0,
                  "y": 0,
                },
              ],
              "name": "testName",
            }
          }
          tableIndex={1}
          tables={
            Array [
              Object {
                "attributes": Array [
                  Object {
                    "field": "testField",
                    "type": "testType",
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "field": "testField2",
                    "type": "testType2",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "name": "testName",
              },
              Object {
                "attributes": Array [
                  Object {
                    "field": "testField",
                    "type": "testType",
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "field": "testField2",
                    "type": "testType2",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "name": "testName",
              },
              Object {
                "attributes": Array [
                  Object {
                    "field": "testField",
                    "type": "testType",
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "field": "testField2",
                    "type": "testType2",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "name": "testName",
              },
            ]
          }
        />
        <Table
          data={
            Object {
              "tables": Array [
                Object {
                  "attributes": Array [
                    Object {
                      "field": "testField",
                      "type": "testType",
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "field": "testField2",
                      "type": "testType2",
                      "x": 0,
                      "y": 0,
                    },
                  ],
                  "name": "testName",
                },
                Object {
                  "attributes": Array [
                    Object {
                      "field": "testField",
                      "type": "testType",
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "field": "testField2",
                      "type": "testType2",
                      "x": 0,
                      "y": 0,
                    },
                  ],
                  "name": "testName",
                },
                Object {
                  "attributes": Array [
                    Object {
                      "field": "testField",
                      "type": "testType",
                      "x": 0,
                      "y": 0,
                    },
                    Object {
                      "field": "testField2",
                      "type": "testType2",
                      "x": 0,
                      "y": 0,
                    },
                  ],
                  "name": "testName",
                },
              ],
            }
          }
          draggable={true}
          onAddRow={[Function]}
          onDragTable={[Function]}
          refreshTableRefs={[Function]}
          style={
            Object {
              "backgroundColor": "FireBrick",
            }
          }
          table={
            Object {
              "attributes": Array [
                Object {
                  "field": "testField",
                  "type": "testType",
                  "x": 0,
                  "y": 0,
                },
                Object {
                  "field": "testField2",
                  "type": "testType2",
                  "x": 0,
                  "y": 0,
                },
              ],
              "name": "testName",
            }
          }
          tableIndex={2}
          tables={
            Array [
              Object {
                "attributes": Array [
                  Object {
                    "field": "testField",
                    "type": "testType",
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "field": "testField2",
                    "type": "testType2",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "name": "testName",
              },
              Object {
                "attributes": Array [
                  Object {
                    "field": "testField",
                    "type": "testType",
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "field": "testField2",
                    "type": "testType2",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "name": "testName",
              },
              Object {
                "attributes": Array [
                  Object {
                    "field": "testField",
                    "type": "testType",
                    "x": 0,
                    "y": 0,
                  },
                  Object {
                    "field": "testField2",
                    "type": "testType2",
                    "x": 0,
                    "y": 0,
                  },
                ],
                "name": "testName",
              },
            ]
          }
        />
      </div>
    </div>
  </Scrollbars>
</div>
`;
